app/models/miq_group.rb

```diff
@@ -230,8 +230,14 @@ def validate_default_tenant
     end
   end
 
+  def own_users
+    own_users = []
+    users.each { |user| own_users << user if user.miq_groups.count == 1 }
+    own_users
+  end
```

Comment:
I believe that we can remove N+1 here by this query:

```ruby
  def users_assigned_to_other_groups?
    group_user_ids = user_ids
    User.includes(:miq_groups).where(:id => group_user_ids).where.not(:miq_groups => {:id => id}).count == group_user_ids.size
  end
```
which means:
select users of the group and elimited the users with the our group(which we want to delete)
and if all (comparing with `== group_user_ids.size`)users will belongs to any other groups than can delete it.

also, I would like to change message to something more descriptive.

```
  raise _("All users don't belong to other groups  ") unless users_assigned_to_other_groups?
```

What do you think?